nova_plugin/server.py

```diff
@@ -760,8 +760,8 @@ def _get_private_key(private_key_path):
         key_path = private_key_path
     else:
         key_path = \
-            pk_node_by_rel.properties['private_key_path'] or \
-            ctx.bootstrap_context.cloudify_agent.agent_key_path
+            (pk_node_by_rel and pk_node_by_rel.properties['private_key_path']) \
```

Comment:
This is correct: (pk_node_by_rel and pk_node_by_rel.properties['private_key_path'])
but the syntax is very confused, please change to something like:
if pk_node_by_rel and pk_node_by_rel.properties['...']:
  key_path = ...
else:
  key_path = ...